storage/remote/client.go

```diff
@@ -138,11 +140,19 @@ func NewWriteClient(name string, conf *ClientConfig) (WriteClient, error) {
 	if err != nil {
 		return nil, err
 	}
-
 	t := httpClient.Transport
+
+	if conf.SigV4Config != nil {
+		t, err = newSigV4RoundTripper(conf.SigV4Config, httpClient.Transport)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
@roidelapluie I rebased and made injected headers happen after the sigv4. I think this is the right way to handle it, but PTAL to double-check.